packages/stream/index.js

```diff
@@ -131,11 +105,11 @@ export const makeStream = (acks, data) => {
 harden(makeStream);
 
 /**
- * @template T
- * @template U
+ * @template TRead
+ * @template TWrite
  * @template TReturn
  * @template UReturn
- * @returns {[Stream<T, U, TReturn>, Stream<U, T, UReturn>]}
+ * @type {import('./types.js').makePipeFn<TRead, TWrite, TReturn, UReturn>}
```

Comment:
I'm thinking that we should instead type the `data` and `acks` queues, and let type inference work through. Combined with a `types.test.ts` file for types, it should be enough to ensure the return type is correct.

Reply:
Alternatively it might be possible to `declare function makePipe` in a `.d.ts` file and do a `@returns {ReturnType<typeof import('./index.js').makePipe<TRead, TWrite, TReturn, UReturn>>}`

Reply:
Please take a look again. It left a note about the error I ran into attempting to get the type system to recognize the entangled queues as producing complementary streams. When I added the external type validation, a bunch of things needed changing, but at the end of the day, the definition of `makePipe` in `index.d.ts` turned out to be useful, and the internal JSDoc definition could not be compelled to cooperate.